### ISSUE TITLE
Error on registration - TimeZone issues

### ISSUE THREAD

**Issue by kimdre** [antedebaas/DMARC-Reports#12]
When I try to register my first user I receive an 500 error on the web UI after I click the "Register" button (Error see below).
It seems like the mailer client isn't able to send the registration mail via SMTP (I tried both ports for secure SMTP, 465 and 587)
I also tried different passwords with different lengths and strengths.

My docker-compose.yml
```yaml
name: dmarc-smtptls-reports
version: "2"
services:
  app:
    image: antedebaas/dmarc-reports:latest
    ports:
        - 8087:8080
    environment:
      TZ: Europe/Berlin
      DATABASE_TYPE: mysql
      DATABASE_HOST: db1.prod.<mydomain>
      DATABASE_PORT: 3306
      DATABASE_NAME: dmarcreports
      DATABASE_USER: dmarcreports
      DATABASE_PASSWORD: '<db_password>'
      MAILER_HOST: mail.<mydomain>
      MAILER_SMTPPORT: 587
      MAILER_IMAPPORT: 993
      MAILER_USER: dmarc-reports@<mydomain>
      MAILER_PASSWORD: '<mailer_password>'
    restart: unless-stopped
```

UI Error
```
Oops! An Error Occurred
The server returned a "500 Internal Server Error".
Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.
```

Docker Log:
```
NOTICE: PHP message: [critical] Uncaught PHP Exception Symfony\Component\Mailer\Exception\TransportException: "Failed to authenticate on SMTP server with username "dmarc-reports@<mydomain>" using the following authenticators: "LOGIN", "PLAIN". Authenticator "LOGIN" returned "Expected response code "235" but got code "535", with message "535 5.7.8 Error: authentication failed: UGFzc3dvcmQ6".". Authenticator "PLAIN" returned "Expected response code "235" but got code "535", with message "535 5.7.8 Error: authentication failed: UGFzc3dvcmQ6"."." at /var/www/html/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php line 225

2023/11/19 22:45:35 [error] 16#16: *7 FastCGI sent in stderr: "PHP message: [critical] Uncaught PHP Exception Symfony\Component\Mailer\Exception\TransportException: "Failed to authenticate on SMTP server with username "dmarc-reports@<mydomain>" using the following authenticators: "LOGIN", "PLAIN". Authenticator "LOGIN" returned "Expected response code "235" but got code "535", with message "535 5.7.8 Error: authentication failed: UGFzc3dvcmQ6".". Authenticator "PLAIN" returned "Expected response code "235" but got code "535", with message "535 5.7.8 Error: authentication failed: UGFzc3dvcmQ6"."." at /var/www/html/vendor/symfony/mailer/Transport/Smtp/EsmtpTransport.php line 225" while reading response header from upstream, client: 172.26.0.1, server: _, request: "POST /register HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "dmarc.<mydomain>", referrer: "https://dmarc.<mydomain>/register"
```

**Comment by antedebaas** [comment]
"authentication failed"
seems an authentication error.
are you sure the email account exists?

**Comment by kimdre** [comment]
Huh weird I recreated the mailbox today and now it is working.
New error: The confirmation token I receive from the email is invalid right at generation because the docker container has no timezone support. Alpine images don't come with the tzdata package preinstalled.

For example this link is generated at 19:39 (GMT) but is invalid when I click on it because the timestamp in the Url was expired at 19:38, which was before the link was even generated:
`http://<mydoman>/verify/email?expires=1700509133&id=1&signature=29R6tZEeZx5xSpzZl5KFrrLukfwrcxQliGj%2BQO5Sxmw%3D&token=wbo%2FzJSZW%2Bvg981qNsTdmCQ6v6wZ3gZXYhqx3jlo9I4%3D`

**Comment by antedebaas** [comment]
the token validation happens in the same container as the token creation.
timezones are thus not relevant here.
Im guessing there is something else going on.
could you try it once more? and if it still does not work ill look into it

**Comment by kimdre** [comment]
I tried it again. Still the same issue.

**Comment by antedebaas** [comment]
I’ll look into it

**Comment by antedebaas** [comment]
The problem seems to appear because you added 
`      TZ: Europe/Berlin`
to the environment in combination with a bug in the verify-email-bundle.
https://github.com/SymfonyCasts/verify-email-bundle/issues/116

This needs to be fixed in the verify-email-bundle

**Comment by kimdre** [comment]
Today I tried it again and now it works!

**Comment by antedebaas** [comment]
@kimdre did you change anything? And if so, could you please share it?

**Comment by jrushlow** [comment]
Howdy @antedebaas - I apologize in advance for commenting on a closed issue. But in regards to:

> The problem seems to appear because you added ` TZ: Europe/Berlin` to the environment in combination with a bug in the verify-email-bundle. [SymfonyCasts/verify-email-bundle#116](https://github.com/SymfonyCasts/verify-email-bundle/issues/116)
> 
> This needs to be fixed in the verify-email-bundle

This appears to be related to the `diff()` timezone bug that existed in early versions of PHP 8.1. See (demonstrates the bug): 
https://github.com/SymfonyCasts/reset-password-bundle/issues/219#issuecomment-1075240706

The actual PHP Bug Report: https://bugs.php.net/bug.php?id=81263

The problem has since been fixed since PHP `8.1.10`. If there are still issues with `verify-email-bundle` && timezone, please do let us know so we can work out a fix.

Thanks! - I hope this helps for anyone that's still on early versions of 8.1 and experiencing this issue.

**Comment by antedebaas** [comment]
@jrushlow Thank you for the information.
the project has since then upped the minimum version requirement to php 8.2 so this should no longer be an issue then.
